kubectl-rook-ceph.sh

```diff
@@ -273,6 +274,90 @@ function run_purge_osd() {
     rook ceph osd remove --osd-ids=$1 --force-osd-removal=$force_removal"
 }
 
+function run_cluster_health() {
+  end_of_command_parsing "$@" # end of command tree
+  set +e                      # if there are no pod that are not running, then do no exit
+  check_mon_pods_nodes
+  echo
+  check_mon_quorum
+  echo
+  check_osd_pod_count_and_nodes
+  echo
+  check_all_pods_status
+  echo
+  check_pg_are_active_clean
+  echo
+  check_mgr_pods_status_and_count
+  set -e
+}
+
+function check_mon_pods_nodes() {
+  echo -e "${INFO_PREFIX} Checking if at least three mon pods are running on different nodes"
+  mon_unique_node_count=$(KUBECTL_NS_CLUSTER get pod | grep mon | grep -v canary | awk '{print $2}' | sort | uniq | wc -l)
+  if [ "$mon_unique_node_count" -lt 3 ]; then
+    echo -e "${WARNING_PREFIX} At least three mon pods should running on different nodes"
+  fi
+  KUBECTL_NS_CLUSTER get pod | grep mon | grep --invert-match canary
+}
+
+function check_mon_quorum() {
+  echo -e "${INFO_PREFIX} Checking mon quorum and ceph health details"
+  ceph_health_details=$(KUBECTL_NS_OPERATOR exec deploy/rook-ceph-operator -- ceph health detail --conf="$CEPH_CONF_PATH")
+  if [[ "$ceph_health_details" = "HEALTH_OK" ]]; then
+    echo -e "$ceph_health_details"
+  elif [[ "$ceph_health_details" =~ "HEALTH_WARN" ]]; then
+    echo -e "${WARNING_PREFIX} $ceph_health_details"
+  elif [[ "$ceph_health_details" =~ "HEALTH_ERR" ]]; then
+    echo -e "${ERROR_PREFIX} $ceph_health_details"
+  fi
+}
+
+function check_osd_pod_count_and_nodes() {
+  echo -e "${INFO_PREFIX} Checking if at least three osd pods are running on different nodes"
+  osd_unique_node_count=$(KUBECTL_NS_CLUSTER get pod | grep osd | grep -v prepare | awk '{print $2}' | sort | uniq | wc -l)
+  if [ "$osd_unique_node_count" -lt 3 ]; then
+    echo -e "${WARNING_PREFIX} At least three osd pods should running on different nodes"
+  fi
+  KUBECTL_NS_CLUSTER get pod | grep osd | grep --invert-match prepare
+}
+
+function check_all_pods_status() {
+  echo -e "${INFO_PREFIX} Pods that are in 'Running' status"
+  KUBECTL_NS_OPERATOR get pod --field-selector status.phase=Running
+  if [ "$ROOK_OPERATOR_NAMESPACE" != "$ROOK_CLUSTER_NAMESPACE" ]; then
+    KUBECTL_NS_CLUSTER get pod --field-selector status.phase=Running
+  fi
+
+  echo
+  echo -e "${WARNING_PREFIX} Pods that are 'Not' in 'Running' status"
```

Comment:
I see a lot of this `echo -e "${SOME_PREFIX} text".

Generally, I have used helpers for things like this

```sh
function info_msg() {
  echo -e "${INFO_PREFIX}""$@"
}

function warn_msg() {
  echo -e "${WARN_PREFIX}""$@"
}

function error_msg() {
  echo -e "${ERROR_PREFIX}""$@"
}
```


Then these messages are called like

```sh
info_msg "checking placement group status"

warn_msg "$pg_state"

error_msg "$pg_state"
```

Reply:
done

Reply:
thanks for the suggestion